libs/ApplicationCore/src/ApplicationCore/UserConfiguration.cpp

```diff
@@ -0,0 +1,45 @@
+#include "UserConfiguration.h"
+
+#include "Conversions.h"
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+namespace Status::ApplicationCore {
+
+/// `status-go` data location
+constexpr auto dataSubfolder = "data";
+
+UserConfiguration::UserConfiguration(QObject *parent)
+    : QObject{parent}
+{
+    generateReleaseConfiguration();
```

Comment:
> Coding rules suggestion: use `this->`for access self methods

As far as I remember team decided for `m_` prefix and never mentioned or encountered the java style `this->`.

> Also if generateReleaseConfiguration() called only in construction it can be replaced with more immutable-style: `UserConfiguration::UserConfiguration(QObject *parent): QObject(parent), m_userDataFolder(StatusPath(::dataSubfolder).writable()` m_userDataFolder should be const for this way

There were initially more paths to initialize and release/debug configurations. I removed this for simplicity. This part will be refactored when the Login is extended and refactored.